File: back-end/src/main/java/CRM4RTONOCTURNOA/CRM/service/CampaniaService.java
package CRM4RTONOCTURNOA.CRM.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import CRM4RTONOCTURNOA.CRM.entity.Campania;
import CRM4RTONOCTURNOA.CRM.repository.CampaniaRepository;

@Service
public class CampaniaService {
    @Autowired
    CampaniaRepository campaniaRepository;
    //Create y Update
    public Campania save(Campania campania){
        return campaniaRepository.save(campania);
    }
    //Read
    public Campania findById(Long id){
        return campaniaRepository.findById(id).get();
    }
    //Delete
    public void deleteById(Long id){
        campaniaRepository.deleteById(id);
    }
    //findAll
    public List<Campania> findAll(){
        return campaniaRepository.findAll();
    }
}

File: fron-end/src/app/campania/campania.service.ts
import { Injectable } from '@angular/core';
import { Campania } from './campania';
import { Observable } from 'rxjs';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from 'src/environments/environment';


@Injectable({
  providedIn: 'root'
})
export class CampaniaService {

  private httpOptions =  {
    headers: new HttpHeaders({"Content-Type":"application/json"})
  }

  private url = environment.API_URL + "/campania";

  constructor(
    private http: HttpClient
  ) { }

  //Create
  public save(campania: Campania): Observable<Campania>
  {
    return this.http.post<Campania>
    (this.url+"/save",campania,this.httpOptions);
  }

  //Read
  public findById(id: number): Observable<Campania>
  {
    return this.http.get<Campania>(this.url+"/id", this.httpOptions);
  }

  //Update
  public update(campania: Campania): Observable<Campania>
  {
    return this.http.put<Campania>(this.url+"/update", campania, this.httpOptions);
  }

  //Delete
  public deleteById(id: number): Observable<Campania>
  {
    return this.http.delete<Campania>(this.url+"/deleteById/"+id, this.httpOptions);
  }

  //(FindAll)
  public findAll(): Observable<Campania[]>
  {
    return this.http.get<Campania[]>(this.url+"/findAll", this.httpOptions);
  }
}
